device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -885,9 +899,9 @@ private void addProxyLayer(Transport transport, String hostName)
     private void sendQueuedMessages()
     {
         int messagesAttemptedToBeProcessed = 0;
-        Message message = messagesToSend.poll();
-        while (message != null && messagesAttemptedToBeProcessed < MAX_MESSAGES_TO_SEND_PER_CALLBACK)
+        while (messagesToSend.peek() != null && messagesAttemptedToBeProcessed < MAX_MESSAGES_TO_SEND_PER_CALLBACK)
         {
+            Message message = messagesToSend.poll();
```

Comment:
Are we threaded/synchronized in anyway here? I like this order of execution but should we check to see if message != null before executing anything below?